RadixWallet/Clients/ProfileStore/ProfileStore.swift

```diff
@@ -1,284 +1,287 @@
+		let idOfEphemeralProfileToDelete = self.profile.networks.isEmpty ? self.profile.id : nil
 
-	/// A multicasting replaying async sequence of distinct Gateways
-	public func gatewaysValues() async -> AnyAsyncSequence<Gateways> {
-		lens {
-			$0.profile.appPreferences.gateways
-		}
-	}
+		var profileToImport = profileToImport
 
-	/// A multicasting replaying async sequence of distinct FactorSources
-	public func factorSourcesValues() async -> AnyAsyncSequence<FactorSources> {
-		lens {
-			$0.profile.factorSources
-		}
-	}
+		// Before saving it we must claim ownership of it!
+		try _claimOwnership(of: &profileToImport)
 
-	public func getLoadProfileOutcome() async -> LoadProfileOutcome {
-		switch self.profileStateSubject.value {
-		case .persisted:
-			.existingProfile
-		case let .ephemeral(ephemeral):
-			if let error = ephemeral.loadFailure {
-				.usersExistingProfileCouldNotBeLoaded(failure: error)
-			} else {
-				.newUser
+		try updateHeaderOfThenSave(
+			profile: profileToImport
+		)
+
+		if let idOfEphemeralProfileToDelete {
+			do {
+				try secureStorageClient.deleteProfileAndMnemonicsByFactorSourceIDs(
+					profileID: idOfEphemeralProfileToDelete,
+					keepInICloudIfPresent: false
+				)
+			} catch {
+				// Not important enought to fail
+				logAssertionFailure("Failed to delete empty ephemeral profile ID, error: \(error)")
 			}
 		}
 	}
 
-	public func importProfileSnapshot(_ profileSnapshot: ProfileSnapshot) async throws {
-		try assertProfileStateIsEphemeral()
-
-		guard await (try? secureStorageClient.loadProfileSnapshotData(profile.header.id)) == nil else {
-			struct ExistingProfileSnapshotFoundAbortingImport: Swift.Error {}
-			throw ExistingProfileSnapshotFoundAbortingImport()
+	public func deleteProfile(
+		keepInICloudIfPresent: Bool,
+		assertOwnership: Bool = true
+	) throws {
+		if assertOwnership {
+			// Assert that this device is allowed to make changes on Profile
+			try _assertOwnership()
 		}
 
 		do {
-			try await changeProfileSnapshot(to: profileSnapshot)
+			userDefaultsClient.removeActiveProfileID()
+			try secureStorageClient.deleteProfileAndMnemonicsByFactorSourceIDs(profile.header.id, keepInICloudIfPresent)
 		} catch {
-			let errorMessage = "Critical failure, unable to save imported profile snapshot: \(String(describing: error))"
-			loggerGlobal.critical(.init(stringLiteral: errorMessage))
-			assertionFailure(errorMessage) // for DEBUG builds we want to crash
-			throw error
+			logAssertionFailure("Error, failed to delete profile or factor source, failure: \(String(describing: error))")
 		}
+
+		let profile = try! Self._tryGenerateAndSaveNewProfile(deviceInfo: deviceInfo)
+		self.profileSubject.send(profile)
 	}
 
-	public func importCloudProfileSnapshot(_ header: ProfileSnapshot.Header) async throws {
-		try assertProfileStateIsEphemeral()
+	public func finishedOnboarding() async {
+		@Dependency(\.device) var device
+		if !profile.hasMainnetAccounts() {
```

Comment:
Note: This will go away with my upcoming PR, where it is not possible to not have an account on a network.